### DISCUSSION TITLE
Make every file in /api a route

### DISCUSSION THREAD

**Discussion by remy** [vercel/vercel#4673] [Help]
I'm sure this has been asked before (and answered, but I couldn't find the answer searching these discussions).

I've made two files in `/api` - `index.js` and `single.js` (and intend to make more).

How do I tell Now to just make every file it's own route (like I do with a Next project)? Or do I need to manually specify a route for every instance of a file?

I've looked here and can't see it (or have missed it): https://vercel.com/docs/configuration#introduction/configuration-reference

Thanks in advance.

**Comment by mcsdevv** [comment]
Hey @remy, by default, the routing is filesystem based - are you experiencing different behavior?

**Reply by remy** [reply]
Yes, definitely different.

If I put index.js in the root of the project, it rendered it as plain text.

If I put it in /api I have to navigate to /api/index in the url.

I want to navigate to / and to /single and have it mapped to the /api directory - without having to specify every individual route. 

Are you saying that this happens by default (I'm pretty sure you're not)?

**Reply by mcsdevv** [reply]
Have you got any other configuration that could be impacting this?

I've tested with a deployment using that structure and all works as expected:

- https://api-test-mu.vercel.app/api/ (renders 'Hello world')
- https://api-test-mu.vercel.app/api/single (renders 'Hello world from single.js')

**Reply by remy** [reply]
Right, which isn't what I'm trying to do - sorry if that wasn't clear.

I want to request myapp.app/single and myapp.app/foo etc etc to map to the api dir. Is _that_ possible without having to individually configure every single route?

**Reply by styfle** [reply]
You could try a wildcard rewrite:

```json
{
  "rewrites": [
    { "source": "(.*)", "destination": "/api$1" }
  ]
}
```

or 

```json
{
  "rewrites": [
    { "source": "/(.*)", "destination": "/api/$1" }
  ]
}
```

https://vercel.com/docs/configuration#project/rewrites
